compiler/luci/import/src/Nodes/CircleUnidirectionalSequenceLSTM.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Import/Nodes/CircleUnidirectionalSequenceLSTM.h"
+
+#include <luci/IR/Nodes/CircleUnidirectionalSequenceLSTM.h>
+
+#include <loco.h>
+
+namespace luci
+{
+
+bool CircleUnidirectionalSequenceLSTMGraphBuilder::validate(const ValidateArgs &args) const
+{
+  if (args.op.inputs.size() != 24)
+    return false;
+
+  return true;
+}
+
+CircleNode *CircleUnidirectionalSequenceLSTMGraphBuilder::build_node(
+    const circle::OperatorT &op, const std::vector<CircleNode *> &inputs, loco::Graph *graph) const
+{
+  auto *node = graph->nodes()->create<CircleUnidirectionalSequenceLSTM>();
+  node->input(inputs.at(0));
+
+  node->input_to_input_weights(inputs.at(1)); // Optional
+  if (auto input_to_input_weights =
+          dynamic_cast<luci::CircleOutputExclude *>(node->input_to_input_weights()))
+  {
+    input_to_input_weights->dtype(loco::DataType::FLOAT32);
+  }
+  node->input_to_cell_weights(inputs.at(2));
+  node->input_to_forget_weights(inputs.at(3));
+  node->input_to_output_weights(inputs.at(4));
+
+  node->recurrent_to_input_weights(inputs.at(5)); // Optional
+  if (auto recurrent_to_input_weights =
+          dynamic_cast<luci::CircleOutputExclude *>(node->recurrent_to_input_weights()))
```

Comment:
as `recurrent_to_input_weights()` is `at(5)->node();`  and we have in `loco`, `loco::Node *arg(uint32_t n)` we can revise this as
```cxx
auto recurrent_to_input_weights = dynamic_cast<luci::CircleOutputExclude *>(node->arg(5));
```
then we can add a loop and change all `if (...) { ... }` to one line with an array having all optional index `[5, 9, 10, ...]`...

something like
```cxx
std::vector<int32_t> optionals = { 5, 9, 10, ... }
for (auto optional : optionals)
{
  if (auto inp = dynamic_cast<luci::CircleOutputExclude *>(node->arg(optional)))
  {
    inp ->dtype(loco::DataType::FLOAT32);  
  }
}
```

Reply:
it would be nice if you could add a comment about the reason why we call `dtype(loco::DataType::FLOAT32);`...

Reply:
Oh, Okay I will fix it!

Reply:
> it would be nice if you could add a comment about the reason why we call `dtype(loco::DataType::FLOAT32);`...

Oh, I forget. This Data type depends on `input_to_output_weights` type. I will fix it.